pierce-portfolio/portfolio/src/main/webapp/script.js

```diff
@@ -52,17 +52,35 @@ function handleResponse(response) {
 
   // response.text() returns a Promise, because the response is a stream of
   // content and not a simple variable.
-  const textPromise = response.text();
+  const jsonPromise = response.json();
 
   // When the response is converted to text, pass the result into the
   // addMessageToDom() function.
-  textPromise.then(addMessageToDom);
+  jsonPromise.then(addMessageToDom);
 }
 
 /** Adds the hello world message to the DOM. */
-function addMessageToDom(message) {
-  console.log('Adding message to dom: ' + message);
+function addMessageToDom(comments) {
+  console.log('Adding message to dom: ' + comments);
 
-  const messageContainer = document.getElementById('message-container');
-  messageContainer.innerText = message;
+  const commentContainer = document.getElementById('comments-container');
+  commentContainer.innerHTML = '';
+  var i;
+  commentLoop:
+  for(i=0; i < comments.length; i++) {
+    if(comments[i]) {
```

Comment:
the if-else seems redundant since there's already a loop to iterate over all available comments